File: soal6.c
#include <stdio.h>

int main() {
    int belanja, hasil;

    printf("Masukkan total belanja: ");
    scanf("%d", &belanja);

    hasil = (belanja / 50) * 50;

    printf("Hasil pembulatan = Rp%d\n", hasil);

    return 0;
}

File: soal6.pas
#include <stdio.h>

int main() {
    char ch;
    int angka;

    printf("Masukkan sebuah karakter: ");
    scanf(" %c", &ch);

    if (ch >= '0' && ch <= '9')
        angka = ch - '0';
    else
        angka = -99;

    printf("Hasil konversi = %d\n", angka);

    return 0;
}